feat(tutorial): add third python tutorial (#1960)
scaleway/docs-content: tutorials/python-for-loops/index.mdx

---
meta:
  title: Getting started with Python for loops, with exercises

>   description: This tutorial shows you how to get started with using for loops in Python, showing syntax, examples and practice exercises.

Reviewer: ```suggestion
  description: This tutorial shows you how to get started with using for loops in Python, showing syntax, examples, and practice exercises
```

content:
  h1: Getting started with Python for loops

>   paragraph: This tutorial shows you how to get started with using for loops in Python, showing syntax, examples and practice exercises.

Reviewer: ```suggestion
  paragraph: This tutorial shows you how to get started with using for loops in Python, showing syntax, examples, and practice exercises
```

> tags: python for loop syntax examples statement

Author: ```suggestion
tags: python for loop syntax examples statement for-loop practice exercises break continue iterate syntax
```

categories:
  - compute
dates:
  validation: 2023-06-28
  posted: 2023-06-28
---

<Message type="requirement">

> If you're coming to this tutorial as a complete Python beginner, make sure you're familiar with the Python concepts covered in our [Python for complete beginners](/tutorials/get-started-python/) tutorial, and with [dictionaries and lists](/docs/tutorials/python-lists-dicts/).

Reviewer: ```suggestion
If you're coming to this tutorial as a complete Python beginner, make sure you are familiar with the Python concepts covered in our [Python for complete beginners](/tutorials/get-started-python/) tutorial, and with [dictionaries and lists](/docs/tutorials/python-lists-dicts/).
```

</Message>

## Introduction

> So you know how to **create variables** in Python, as well as how to **create lists** and **dictionaries** to store multiple items in a single variable? Then you're ready to learn about **loops**.

Reviewer: ```suggestion
So you know how to **create variables** in Python, as well as how to **create lists** and **dictionaries** to store multiple items in a single variable? Then you are ready to learn about **loops**.
```

A **loop** is a piece of code that tells Python to repeat a specified set of actions multiple times. Loops are a fundamental part of most programming languages, and let you achieve a whole lot with just a few lines of code. Often, they **iterate** over collections of data (such as data held in dictionaries or lists) and carry out all sorts of operations on those collections.

> Why are loops useful? Let's look at a quick example:

Reviewer: ```suggestion
Why are loops useful? Let us look at a quick example:
```

not sure about this one though

Reviewer 2: `Let's` is okay, the uncontracted form would be unnecessarily formal :)

Reviewer 2: > not sure about this one though

`Let's ` is okay, the uncontracted form would be unnecessarily formal :)

Author: I agree, I'll leave it like it is

Imagine you have a list of numbers, and you want to find every number higher than 10. Without a loop, the code is very long and repetitive:

```python
my_list = [3, 10.1, 62, 9]
  
if my_list[0] > 10:
  print(my_list[0])
  
if my_list[1] > 10:
  print(my_list[1])
  
if my_list[2] > 10:
  print(my_list[2])
  
if my_list[3] > 10:
  print(my_list[3])
  
## Output:
10.1
62
```

With a for loop, we can do it in just three lines of code:

```python
for number in my_list:
  if number > 10:
    print(number)
  
## Output:
10.1
62
```

> Let's dive into the world of for loops in Python, and learn how to code them and what we can do with them.

Reviewer: same as above - Let us?

        3
        ```
      </TabsTab>
    </Tabs>

Ingredients of a for loop (see the syntax above):

- **for**: this keyword is required to start a loop
- **element**: after the `for` keyword, we can use any variable name we want! This variable will simply access each item of the sequence on each iteration. In the examples above you can see that we've called it variously `element`, `color`, `letter`, `product`, and `x`. Anything works, but generally you should try to give this variable a meaningful name: it will make your code easier to read for yourself and anyone else who looks at it.
- **in**: this keyword is required to introduce the object you want to iterate over in the loop
- **iterable-object**: after the `in` keyword, state the object you want to iterate over. This can be any iterable object, including lists, strings, dictionaries, ranges, tuples and sets. 
- The for statement must end with a **colon**, followed by a **line return** and **indentation**.
- **statement(s)**: One or more actions to be carried out on every iteration of the loop. In the examples above we simply use a print statement, but anything is possible.

### Basic exercises

    <Message type="tip">

>     When working with loops, you will probably find it more practical to use a [text editor and the terminal](/tutorials/get-started-python/#6:-from-the-shell-to-a-script) or [Jupyter notebook](https://jupyter.org/)/a , rather than the [Python Interactive Shell](/tutorials/get-started-python/#2:-getting-ready-to-code:-opening-the-python-interactive-shell).

Reviewer: ```suggestion
    When working with loops, you will probably find it more practical to use a [text editor and the terminal](/tutorials/get-started-python/#6:-from-the-shell-to-a-script) or [Jupyter notebook](https://jupyter.org/)/, rather than the [Python Interactive Shell](/tutorials/get-started-python/#2:-getting-ready-to-code:-opening-the-python-interactive-shell).
```

This function can be used in a number of ways:

With just one parameter, the range starts at a default of 0 and finishes at (not including) the specified number:

```python
for x in range(3):
  print(x)
  
## Output:
0
1
2
```

With two parameters, we set the start and finish numbers of the range:

```python
for x in range(2, 6):
  print(x)
  
## Output:
2
3
4
5
```

With three parameters, we set the start and finish numbers of the range, and the **step** value (the interval between each iteration):

```python
for x in range(2, 10, 2):
  print(x)
  
## Output:
2
4
6
8
```

### Range exercises

1. Create a loop that iterates over a range and produces the following output:

    ```
    5
    6
    7
    8
    9
    ```

    Answer [below](#range-exercises:-answers).
  
2. Create a loop that iterates over a range and produces the following output:

    ```
    0
    25
    50
    75
    ```

    Answer [below](#range-exercises:-answers).

3. Create code that includes a loop that iterates over a range, and produces the following output:

    ```
    Counting down...
    5
    4
    3
    2
    1
    Go!
    ```

    Answer [below](#range-exercises:-answers).

> ## Nested for-loops

Reviewer 2: ```suggestion
## Nested for-loops
```

for loops or for-loops? As someone who isn't familiar with Python, I'd lean towards `for-loops` but it seems `for loops` is more common (and preferable SEO wise?)

Author: Hmm, I checked and thought that for-loops seemed more common, but I'll have another look
